views/js/runner/proxy/cache/proxy.js

```diff
@@ -63,27 +63,37 @@ define([
             var self = this;
 
             //we keep items here
-            this.itemStore    = itemStoreFactory(cacheSize);
-
-            //keep track of the calls to prevent doing it again on navigating back and forward
-            this.nextCalledBy = {};
+            this.itemStore = itemStoreFactory(cacheSize);
 
             //can we load the next item from the cache/store ?
-            this.getItemFromStore  = false;
-
-            //check whether we're on the last item
-            this.isLastItem   = false;
+            this.getItemFromStore = false;
 
             //keep a ref to the promise of the loadNextItem in case the call is not done when moving
             this.loadNextPromise = Promise.resolve();
 
+            //preload at least this number of items
+            this.cacheAmount = 1;
+
+            //maintain a map of the items chain: for each item, a link to its neighbors
+            this.itemChain = {};
```

Comment:
This is the point that should be reviewed. The next step will be to work with a dynamic testMap (adaptive session) and the item chain can be changed. I see that's more convenient to maintain such reference, but it will be easier for the future test cases to just read the testMap. So we will have a unique source of data. Something like : 

```js
/**
 * @param {Object} testMap
 * @param {Number} itemPosition - (could be also the item id)
 * @param {String} direction - previous/next/both
 * @param {Number} [size=3] - will be 2xsize if direction is both
 * @returns {Object[]} the collections of items
 */ 
function navigationHelper.getSiblings(testMap, itemPosition, direction, size)
```
No ?